web/src/routes/OrganizationsList/Organization/Tabs/Settings/Settings.tsx

```diff
@@ -7,47 +7,158 @@ import {
   FlexItem,
   FormGroup,
   Form,
+  FormAlert,
   TextInput,
   FormSelect,
   FormSelectOption,
   ActionGroup,
   Button,
+  Alert,
+  AlertGroup,
 } from '@patternfly/react-core';
 import {useLocation} from 'react-router-dom';
 import {useCurrentUser} from 'src/hooks/UseCurrentUser';
 import {useOrganization} from 'src/hooks/UseOrganization';
+import {useOrganizationSettings} from 'src/hooks/UseOrganizationSettings';
+import {IOrganization} from 'src/resources/OrganizationResource';
+import {humanizeTimeForExpiry, getSeconds, isValidEmail} from 'src/libs/utils';
+import {addDisplayError} from 'src/resources/ErrorHandling';
+
+type validate = 'success' | 'warning' | 'error' | 'default';
+const timeMachineOptions = {
+  '0 s': 'a few seconds',
+  '1 d': 'a day',
+  '1 w': '7 days',
+  '2 w': '14 days',
+  '4 w': 'a month',
+};
 
 const GeneralSettings = (props: GeneralSettingsProps) => {
   const location = useLocation();
   const organizationName = props.organizationName;
-
-  const {user} = useCurrentUser();
+  const {user, loading: isUserLoading} = useCurrentUser();
   const {organization, isUserOrganization, loading} =
     useOrganization(organizationName);
+  const [error, setError] = useState<string>('');
+
+  const {updateOrgSettings} = useOrganizationSettings({
+    name: props.organizationName,
+    onSuccess: (result) => {
+      setAlerts(prevAlerts => {
+        return [...prevAlerts,
+          <Alert key="alert" variant="success" title="Successfully updated settings" isInline={true} timeout={5000} />
+        ]
+      });
+    },
+    onError: (err) => {
+      setAlerts(prevAlerts => {
+        return [...prevAlerts,
+          <Alert key="alert" variant="danger" title={err.response.data.error_message} isInline={true} timeout={5000} />
+        ]
+      });
+    },
+  });
+
+  const [alerts, setAlerts] = useState<ReactNode[]>([]);
 
   // Time Machine
-  const timeMachineOptions = ['1 week', '1 month', '1 year', 'Never'];
   const [timeMachineFormValue, setTimeMachineFormValue] = useState(
-    timeMachineOptions[0],
+    timeMachineOptions['0 s'],
```

Comment:
`0s` is actually disabled on quay.io. This default should be read in from the quay config